javascript/jsx/JavaScriptLexer.g4

```diff
@@ -189,6 +185,8 @@ StringLiteral:                 ('"' DoubleStringCharacter* '"'
              |                  '\'' SingleStringCharacter* '\'') {this.ProcessStringLiteral();}
              ;
 
+LinkLiteral: ('http' | 'https' | 'ftp' | 'file') '://' [a-zA-Z0-9./?=]+; // TODO Could be more precise
```

Comment:
LinkLiteral url scheme needn't be precise, we can't enumerate every possible url scheme.

```html
<a href=mailto:example@example.com>e mail</a>
```
<a href=mailto:example@example.com>e mail</a>

Reply:
Yes, that is true, but there are maybe more protocols needed to that list. I just added the default ones I could think of on the top of my head and there are URLs which have `@` which I just remembered so I want to point out that it is known that the pattern is not perfect.